highfidelity/hifi
# Feat/JS-Baker

This PR leverages the new auto-baking in the Asset Server to bake JS files by minifying them.

Testing Notes:
1) Run the built Interface, Domain Server and Assignment Client
2) Launch the asset server and upload a JavaScript file
3) Observe that JS file appears as bakeable and its baked status is toggleable
4) Assignment Client logs should also reflect that JS baking has been performed. (PFA Sample logs) 
5) Launch the Scripting Console and run the minified JS file present in the Asset Server using
   command like Script.include('atp:/butterflies.js');
   a)Observe that the baked script is functionally equivalent to its original counterpart.
   b)When we run the script, observe that assignment client logs indicate that the call for the script 
     has been redirected to the baked script.
6) Observe the unbaked files present in Asset Server should start baking on the launch of asset server.

Note: A pre-existing bug with the JavaScript Console may show up where the console freezes due to subsequent Script.include() commands. I have made sure it is not due to this PR. 

[sampleLogs.txt](https://github.com/highfidelity/hifi/files/1341776/sampleLogs.txt)
